SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -245,10 +245,17 @@ - (void)sdl_stopManager:(BOOL)shouldRestart {
     // Apple Bug ID #30059457
     __weak typeof(self) weakSelf = self;
     dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.5 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
-        [weakSelf.delegate managerDidDisconnect];
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        if (!strongSelf) {
+            return;
+        }
+
+        [strongSelf.delegate managerDidDisconnect];
 
         if (shouldRestart) {
-            [weakSelf.lifecycleStateMachine transitionToState:SDLLifecycleStateStarted];
+            dispatch_async(strongSelf.lifecycleQueue, ^{
```

Comment:
It looks like some of these are wrapped and others are not. I'm assuming that's because you only wrapped the ones that needed it because you were seeing them run off the appropriate queue.

I'm wondering if there's a better way to do this:

1. Create a private method on this class like `sdl_transitionToState:` that just calls the transition on the `lifecycleStateMachine` wrapped in the `dispatch_async`. It seems like that would be more future proof.

2. Add an initializer to `SDLStateMachine` that takes a `dispatch_queue_t` and auto runs all transitions on that queue?